crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -2516,7 +2517,7 @@ impl TwoQubitControlledUDecomposer {
                         .collect::<SmallVec<_>>();
                     Ok((Some(inv_gate.0), inv_gate_params, qubits))
                 }
-                RXXEquivalent::CustomPython(gate_cls) => {
+                RXXEquivalent::CustomPython(gate_cls) => Python::with_gil(|py: Python| {
```

Comment:
I'm fine with this changes since longer term we'll need to do this when running this in a parallel context. But I'm curious what prompted these changes?

Reply:
See the discussion here: https://github.com/Qiskit/qiskit/pull/13568#discussion_r1918690246
I wanted to have a purely-rust `new_inner` function, so I copied the solution from your PR :)